src/views/Swap/SwapGuru/SwapGuru.tsx

```diff
@@ -151,51 +157,133 @@ const useStyles = makeStyles(theme => ({
+    return state.account.balances && state.account.balances.ohm;
+  });
+
+  const pendingTransactions = useSelector((state: any) => {
+    return state.pendingTransactions;
+  });
 
   const setMax = () => {
-    setModel({ pay: "10", receive: "10" });
+    setModel(currentState => ({ ...currentState, pay: ohmBalance, receive: ohmBalance }));
   };
 
   const swapGuru = async () => {
     let swapTx;
+    let isError;
     try {
       const signer = provider.getSigner();
 
-      const swapContract = new ethers.Contract(TangibleTokenSwap.address, TangibleTokenSwap.abi, signer);
-      const swapAmount = ethers.utils.parseUnits("2", "gwei");
+      const swapContract = new ethers.Contract(
+        addresses[chainID].PASSIVE_INCOME_NFT_SWAP,
+        PassiveIncomeNFTSwap,
+        signer,
+      );
 
-      console.log("swapContract", swapContract);
-      console.log("swapAmount", swapAmount);
+      const swapAmount = ethers.utils.parseUnits(model.pay, "gwei");
+      swapTx = await swapContract.swapGURU(swapAmount, model.lockPeriod, onlyLock);
 
-      swapTx = await swapContract.swap(swapAmount);
-
-      console.log("swapTx", swapTx);
-
-      //   dispatch(
-      //     fetchPendingTxns({
-      //       txnHash: swapTx.hash,
-      //       type: "swap_guru",
-      //     }),
-      //   );
+      dispatch(
+        fetchPendingTxns({
+          txnHash: swapTx.hash,
+          text: "Swaping GURU to NFTs",
+          type: "guru_swap",
+        }),
+      );
 
       await swapTx.wait();
     } catch ({ message }) {
-      // @ts-ignore
-      //   setError({ message });
+      isError = true;
+      setOnlyLock(true);
+      dispatch(error("The multiplier not being available for the amount of tokens"));
       console.log("message", message);
     } finally {
+      if (swapTx) {
+        dispatch(clearPendingTxn(swapTx.hash));
+      }
+
+      if (!isError) {
+        setIsSwapPhase(false);
+        setOnlyLock(false);
+      }
     }
   };
 
+  const onApproval = async () => {
+    let approveTx = null;
+    let isError = false;
+    try {
+      const signer = provider.getSigner();
+      const guru_address = addresses[chainID].GURU_ADDRESS;
+      const guruContract = new ethers.Contract(guru_address, PassiveIncomeNFT, signer);
+
+      approveTx = await guruContract.approve(guru_address, ethers.constants.MaxUint256);
+
+      dispatch(
+        fetchPendingTxns({
+          txnHash: approveTx.hash,
+          text: "Approving GURU Swap",
+          type: "guru_swap",
+        }),
+      );
+
+      await approveTx?.wait();
+    } catch (err) {
+      isError = true;
+      dispatch(error("There was an error approving the transaction, please try again"));
+      console.log("error", err);
+    } finally {
+      if (!isError) {
+        setIsSwapPhase(true);
+      }
+      if (approveTx) {
+        dispatch(clearPendingTxn(approveTx.hash));
+      }
+    }
+  };
+
+  const onSubmit = async () => {
```

Comment:
Would wrap any async/await with try/catch, expect in cases they are already handled within awaited methods